js/src/lib/hooks/useCountdownToast.tsx

```diff
@@ -0,0 +1,119 @@
+import { UseToastOptions, useToast, ToastId } from "@chakra-ui/react";
+import { useCallback, useEffect, useMemo, useRef } from "react";
+
+const COUNTDOWN_CONFIG = {
+  TOAST_ID: "lifetime-countdown",
+  WARNING_THRESHOLD: 60, // seconds before expiry to show warning
+  UPDATE_INTERVAL: 1000, // milliseconds
+  MESSAGE: (seconds: number) => `The server will be closed in ${seconds} seconds.`,
+  STYLE: {
+    fontFamily: "monospace",
+  },
+} as const;
+
+interface CountdownToastOptions {
+  position?: UseToastOptions["position"];
+  variant?: UseToastOptions["variant"];
+  status?: UseToastOptions["status"];
+}
+
+/**
+ * Hook to manage countdown toast notifications for server lifetime
+ * @param lifetimeExpiredAt - Date when the server will expire
+ * @param options - Optional toast UI configuration
+ */
+export function useCountdownToast(lifetimeExpiredAt?: Date, options: CountdownToastOptions = {}) {
```

Comment:
In semantics, I think we can change the type annotation to `lifetimeExpiredAt: Date | undefined`
Because we always pass `lifetimeExpiredAt` to the `useCountdownToast`.